### PR TITLE
Product list

### PR DESCRIPTION
**Pull request by obliviusm**
https://github.com/marchi-martius/FreemarketRn/issues/17

### CODE REVIEW

**Comment by obliviusm** [comment]
screenshot_1494631846

**Comment by obliviusm** [comment]
WIP. only UI yet.
working on server-redux thing

File: app/components/CategoryOverviewView.js [hunk 1]
```diff
@@ -3,10 +3,12 @@ import PropTypes from 'prop-types';
 import { View } from 'react-native';
 
 import CategoryHeaderAndroid from './CategoryHeader.android';
+import ProductList from '../containers/ProductList';
```
Comment (marchi-martius) [inline app/components/CategoryOverviewView.js]: Please avoid such hierarchy (pulling containers from other's container's component - view).
We should move all `ProductList` mapping to `CategoryOverview` and pass props downstream.

File: app/components/__tests__/CategoryOverviewView.js [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
 
 it('renders correctly', () => {
+  const store = configureStore()({ products: { list: { records: [] } } });
   const tree = renderer.create(
-    <CategoryOverviewView
-      category={categories[0]}
-      platform="android"
-      onHeaderIconClicked={() => {}}
-    />,
+    <Provider store={store}>
```
Comment (marchi-martius) [inline app/components/__tests__/CategoryOverviewView.js]: My suggestion above will simplify such cases. BTW components should know nothing about store.

File: app/containers/ProductList.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+const mapStateToProps = state => ({
+  products: state.products.list.records,
+});
+
+const mapDispatchToProps = dispatch => ({
+  loadProducts: () => dispatch(requestProducts()),
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(ProductListView);
```
Comment (marchi-martius) [inline app/containers/ProductList.js:14]: Take a look at my comments above about this container.

File: app/sagas/products.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+export function* fetchProducts() {
+  try {
+    const response = yield call(fetchAllProducts);
+    const [, ...products] = response.data;
+    yield put(productsSuccess(products));
+  } catch (e) {
+    yield put(productsError(e));
+  }
+}
```
Comment (marchi-martius) [inline app/sagas/products.js:18]: We should extract code duplication for fetching API. I'll open an Issue for this.

**Review by marchi-martius — CHANGES_REQUESTED** [review]

File: app/sagas/products.js (hunk 4 above)
Comment (marchi-martius) [reply app/sagas/products.js:18]: #26

File: app/components/CategoryOverviewView.js (hunk 1 above)
Comment (obliviusm) [reply app/components/CategoryOverviewView.js]: @marchi-martius I see your point. But now I think that `CategoryOverviewView` doesn't give any UI. So it's not a component. We just need it to hold next containers. So I would rather move mapping below `CategoryOverviewView`, not above.

**Comment by obliviusm** [comment]
@marchi-martius please, take look
Removed `CategoryOverview` for now.
Simplified platform code by https://facebook.github.io/react-native/docs/platform-specific-code.html#platform-specific-extensions

**Review by marchi-martius — APPROVED** [review]

**Pull request merged**